### PR TITLE
everything we need for client to client AS calls

### PR DESCRIPTION
**Pull request by dejfcz**
new interface - clAppService
new conversation status for non-customer clients - Verified
verification process for non-customer clients (VerifyIdentity request/response)

GetIdentityOnlineInformation request/response - information about identity and its application services

ApplicationService Add/Remove request/response - support for application service announcements

CallIdentityApplicationService request/response - call to identity via application service

IncomingCallNotification request/response - push notification node to callee about new call

ApplicationServiceSendMessage request/response - message over clAppService

### CODE REVIEW

**Comment by dejfcz** [comment]
@hellwolf @wigy-opensource-developer @furszy @Luis-Fernando-Molina

**Comment by dejfcz** [comment]
@hellwolf @wigy-opensource-developer @furszy @Luis-Fernando-Molina
I've added more comments and diagrams to explain the usage of the new messages.

File: IopHomeNode.proto3 [hunk 1]
```diff
@@ -248,6 +248,12 @@ message ConversationResponse {
 //
 // An administrative message to test whether the peer is there and responds to requests.
 //
+// Each identity client is required to send a message not more than 60 seconds after the connection
```
Comment (wigy-opensource-developer) [inline IopHomeNode.proto3:263]: Bikeshed moment... What would happen if we raise the timeout to 300s also on the identity clients? Would there be some resource used up too long on the home node? I understand that the chance of losing the connection to a mobile client is higher than to another home node, but is it a big problem if we do not see that immediately, especially if there is no traffic right now?

Battery usage might improve if the mobile clients would not need to send too many packets when their users are idle.

Comment (dejfcz) [reply IopHomeNode.proto3:263]: These numbers are according to @furszy - his reasoning was as you wrote - the mobile devices are less reliable in closing the connections. It has been argued that sending a couple of packets every minute is not a big deal.

These numbers do affect the resources consumed on the node. If a malicious client opens a connection and does not close it, there is a one less legitimate client that the node can effectively serve (so this is a minor DoS prevention as well). There are resources consumed by the open TCP connection itself as well as some resources that the node's implementation uses for each connected client. If there are just tens of those connections, it is not a big deal. If there are thousands, it is a problem.

If these numbers are not reasonable, please discuss with @furszy and upon mutual agreement, please suggest a better alternative.

**Review by wigy-opensource-developer — APPROVED** [review]
I have no idea how complex the code would get if we would allow checking in multiple services in a single request. If the round-trip-time is big, it could save some seconds on startup. Of course if the time spent by the node creating the response is comparable to the round-trip-time, we would not gain much and then I would stick to this 1 checkin - 1 request protocol.

**Comment by dejfcz** [comment]
Application services check-in is just an announcement of the client that it support certain set of services in this session. At this moment, the protocol allows announcing any number of AS (which I will limit in next commit to 50) at once and deleting AS from the set one by one. If a client wants to check AS one by one, it certainly can.

**Comment by dejfcz** [comment]
For some reason this PR was closed (by me?!), continue in #5

**Pull request closed without merging**